eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1629,43 +1596,44 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder, val
       // if channel is private, we send the channel_update directly to remote
       // they need it "to learn the other end's forwarding parameters" (BOLT 7)
       (state, nextState, stateData, nextStateData) match {
-        case (_, _, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
+        case (NORMAL | SYNCING, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
```

Comment:
🔍 🕵️‍♂️ 

We only update the `buried` boolean in `NORMAL` state:

```suggestion
        case (NORMAL, NORMAL, d1: DATA_NORMAL, d2: DATA_NORMAL) if !d1.commitments.announceChannel && !d1.buried && d2.buried =>
```

Reply:
Wait, something is fishy. Indeed we never update the `channelUpdate` in `SYNCING`, and we don't always update it to `disabled` when going from `NORMAL`->`OFFLINE` (see [below](https://github.com/ACINQ/eclair/pull/2228#discussion_r845118794)). How can we be sure that `d2.channelUpdate` is `enabled` in the [middle handler](https://github.com/ACINQ/eclair/blob/eaa488b59a308307080e71689782d19899c21567/eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala#L1602)? The comment seems inaccurate.

Reply:
You're right, there is a potential issue here (unrelated to the changes in this PR). If the channel has been disabled without changing the `channel_update`, when we come back online we will send the update with the `disabled` bit set.

I don't think it's a huge issue though, because this is a private channel: our peer only uses the fees/cltv to create invoices, they most likely ignore the `disabled` bit. But we could also check it here and avoid sending anything if the update is still disabled (so we will only send a channel update in the next `NORMAL -> NORMAL` transition).